[BUILD] Wrong boost sha1.hpp include in src/preferences.cpp

```
Building: src/preferences.cpp
src/preferences.cpp:31:10: fatal error: boost/uuid/sha1.hpp: No such file or directory
 #include <boost/uuid/sha1.hpp>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [Makefile:236: build/preferences.o] Error 1

```

For modern Boost 1.68 the correct path to include is boost/uuid/detail/sha1.hpp

I can start a PR on that but I'm not sure if changing this path won't affect users with other Boost versions.
It should be integrated somehow in makefile scripts probably.

--- a/src/formula_function.cpp
+++ b/src/formula_function.cpp
@@ -29,17 +29,13 @@
 #include <boost/format.hpp>
 #include <boost/lexical_cast.hpp>
 
-// TODO: when anura boost req >=1.66 remove deprecated path
-#if __has_include("boost/uuid/sha1.hpp")
-
-#include <boost/uuid/sha1.hpp>
-
+#if defined __has_include
+#  if __has_include("boost/uuid/detail/sha1.hpp")
+#    include <boost/uuid/detail/sha1.hpp>
+#  endif
 #else
-
-#include <boost/uuid/detail/sha1.hpp>
-
+#  include <boost/uuid/sha1.hpp>
 #endif
-
 
 #include <boost/algorithm/string.hpp>
 #include <boost/exception/diagnostic_information.hpp>

--- a/src/formula_function.cpp
+++ b/src/formula_function.cpp
@@ -32,6 +32,8 @@
 #if defined __has_include
 #  if __has_include("boost/uuid/detail/sha1.hpp")
 #    include <boost/uuid/detail/sha1.hpp>
+#  else
+#    include <boost/uuid/sha1.hpp>
 #  endif
 #else
 #  include <boost/uuid/sha1.hpp>

--- a/src/formula_function.cpp
+++ b/src/formula_function.cpp
@@ -28,7 +28,19 @@
 
 #include <boost/format.hpp>
 #include <boost/lexical_cast.hpp>
+
+// TODO: when anura boost req >=1.66 remove deprecated path
+#if __has_include("boost/uuid/sha1.hpp")
+
 #include <boost/uuid/sha1.hpp>
+
+#else
+
+#include <boost/uuid/detail/sha1.hpp>
+
+#endif
+
+
 #include <boost/algorithm/string.hpp>
 #include <boost/exception/diagnostic_information.hpp>
 #include <boost/exception_ptr.hpp>

--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -30,17 +30,12 @@
 #include <boost/lexical_cast.hpp>
 
 #if defined __has_include
-#if __has_include("boost/uuid/sha1.hpp")
-
-#include <boost/uuid/sha1.hpp>
-
-#endif
+#  if __has_include("boost/uuid/detail/sha1.hpp")
+#    include <boost/uuid/detail/sha1.hpp>
+#  endif
 #else
-
-#include <boost/uuid/detail/sha1.hpp>
-
+#  include <boost/uuid/sha1.hpp>
 #endif
-
 
 #include "asserts.hpp"
 #include "controls.hpp"

--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -28,7 +28,18 @@
 
 #include <boost/regex.hpp>
 #include <boost/lexical_cast.hpp>
+
+// TODO: when anura boost req >=1.66 remove deprecated path
+#if __has_include("boost/uuid/sha1.hpp")
+
 #include <boost/uuid/sha1.hpp>
+
+#else
+
+#include <boost/uuid/detail/sha1.hpp>
+
+#endif
+
 
 #include "asserts.hpp"
 #include "controls.hpp"

--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -29,11 +29,12 @@
 #include <boost/regex.hpp>
 #include <boost/lexical_cast.hpp>
 
-// TODO: when anura boost req >=1.66 remove deprecated path
+#if defined __has_include
 #if __has_include("boost/uuid/sha1.hpp")
 
 #include <boost/uuid/sha1.hpp>
 
+#endif
 #else
 
 #include <boost/uuid/detail/sha1.hpp>

--- a/src/preferences.cpp
+++ b/src/preferences.cpp
@@ -32,6 +32,8 @@
 #if defined __has_include
 #  if __has_include("boost/uuid/detail/sha1.hpp")
 #    include <boost/uuid/detail/sha1.hpp>
+#  else
+#    include <boost/uuid/sha1.hpp>
 #  endif
 #else
 #  include <boost/uuid/sha1.hpp>

